journal/src/main/scala/com/evolutiongaming/kafka/journal/JsonCodec.scala

```diff
@@ -16,7 +16,7 @@ final case class JsonCodec[F[_]](
 
 object JsonCodec {
 
-  def apply[F[_]](implicit F: JsonCodec[F]): JsonCodec[F] = F
+  def summon[F[_]](implicit F: JsonCodec[F]): JsonCodec[F] = F
```

Comment:
@lavrov @t3hnar Why it is renamed to `summon` instead of `apply`? I think summoning through `apply` is universally accepted practice.

Reply:
We agreed with @lavrov to start moving away from use of `apply` method for "summoning" implicit

- we'd better reserve `apply` as factory method
- `summon` is reserved keyword for exact this purpose in scala 3

Reply:
I have googled a bit, and I could not find any Scala 2 project that uses `summon` in such a way. It means, if we go that way, we will be pioneers and will confuse every single new developer a bit. Not in a critical way, `summon` is quite a clear and well chosen name, but there will be a long period of time when both `apply` and `summon` are used in our projects. Are we sure we want to be pioneers in that area?

> we'd better reserve apply as factory method
I think we are not breaking the commonplace `apply` contract anyhow when using it for summoning (we are returning instance of the class after all), and we can still use `apply` for factory methods even if we used it for summoning, I think, or not?

Reply:
> Are we sure we want to be pioneers in that area?

I personally sure, using `apply` for that have some drawbacks

Reply:
@t3hnar I agree about drawbacks, but it just came to be that we might be breaking the expectations of most typeclass libraries, including, but not limited to:
- https://typelevel.org/cats-tagless/
- https://github.com/typelevel/simulacrum

I think we need to be careful here. I see that some libraries do not require `apply`, i.e. this one, so it might still be okay, but I am a bit worried:
https://github.com/augustjune/context-applied